website/docs/reference/transforms/filter.md

```diff
@@ -0,0 +1,310 @@
+  templateable={false}
+  type={"string"}
+  unit={null}
+  >
+
+#### `[field-name]`.eq
+
+Check whether a fields contents exactly matches the value specified.
+
+
+
+
+</Field>
+
+
+<Field
+  common={false}
+  defaultValue={null}
+  enumValues={null}
+  examples={[{"host.exists":true}]}
+  groups={[]}
+  name={"`[field-name]`.exists"}
+  path={"condition"}
+  relevantWhen={{"type":"check_fields"}}
+  required={false}
+  templateable={false}
+  type={"bool"}
+  unit={null}
+  >
+
+#### `[field-name]`.exists
+
+Check whether a field exists or does not exist, depending on the provided value
+being `true` or `false` respectively.
+
+
+
+
+</Field>
+
+
+<Field
+  common={false}
+  defaultValue={null}
+  enumValues={null}
+  examples={[{"method.neq":"POST"}]}
+  groups={[]}
+  name={"`[field-name]`.neq"}
+  path={"condition"}
+  relevantWhen={{"type":"check_fields"}}
+  required={false}
+  templateable={false}
+  type={"string"}
+  unit={null}
+  >
+
+#### `[field-name]`.neq
+
+Check whether a fields contents does not match the value specified.
+
+
+
+
+</Field>
+
+
+<Field
+  common={true}
+  defaultValue={null}
+  enumValues={null}
+  examples={[{"message.contains":"foo"}]}
+  groups={[]}
+  name={"`[field_name]`.contains"}
+  path={"condition"}
+  relevantWhen={{"type":"check_fields"}}
+  required={false}
+  templateable={false}
+  type={"string"}
+  unit={null}
+  >
+
+#### `[field_name]`.contains
+
+Checks whether a string field contains a string argument.
+
+
+
+
+</Field>
+
+
+<Field
+  common={true}
+  defaultValue={null}
+  enumValues={null}
+  examples={[{"environment.prefix":"staging-"}]}
+  groups={[]}
+  name={"`[field_name]`.prefix"}
+  path={"condition"}
+  relevantWhen={{"type":"check_fields"}}
+  required={false}
+  templateable={false}
+  type={"string"}
+  unit={null}
+  >
+
+#### `[field_name]`.prefix
+
+Checks whether a string field has a string argument prefix.
+
+
+
+
+</Field>
+
+
+</Fields>
+
+</Field>
+
+
+</Fields>
+
+## Output
+
+The `filter` transform accepts and [outputs `log` events](#output) allowing you to select events based on a set of logical conditions.
+For example:
+
+
+The `filter` transform is a simple conditional match, forwarding only those messages that pass all the conditions.
+In this example, we drop all events that do not come from the host `gerry`:
+
+<CodeHeader fileName="vector.toml" />
+
+```toml
+[transforms.from_gerry]
+  inputs = [ "somewhere" ]
+  type = "filter"
+
+  [transforms.from_gerry.condition]
+    "host.eq" = "gerry"
+
+[sinks.only_gerry]
+  inputs = [ "from_gerry" ]
+  type = "something"
+```
+
+Any event that does not match all of the conditions in the filter will be dropped by the transform.
+
+## How It Works
+
+### Complex Processing
+
+If you encounter limitations with the `filter`
+transform then we recommend using a [runtime transform][urls.vector_programmable_transforms].
+These transforms are designed for complex processing and give you the power of
+full programming runtime.
```

Comment:
Perhaps a section here guiding users to swimlanes if they need to split based on a condition instead of just reject?